types/validator.go

```diff
@@ -46,8 +46,9 @@ func (v *Validator) ValidateBasic() error {
 		return errors.New("validator has negative voting power")
 	}
 
-	if len(v.Address) != crypto.AddressSize {
-		return fmt.Errorf("validator address is the wrong size: %v", v.Address)
+	addr := v.PubKey.Address()
+	if !bytes.Equal(v.Address, addr) {
+		return fmt.Errorf("validator address is incorrectly derived from pubkey. Exp: %v, got %v", addr, v.Address)
```

Comment:
fwiw to other reviewers, here's the fix with statesync

https://github.com/cometbft/cometbft/security/advisories/GHSA-g5xx-c4hv-9ccc